terraform/dotnet/eks/main.tf

```diff
@@ -0,0 +1,310 @@
+  host                   = data.aws_eks_cluster.testing_cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.testing_cluster.certificate_authority[0].data)
+  token                  = data.aws_eks_cluster_auth.testing_cluster.token
+  load_config_file       = false
+}
+
+data "template_file" "kubeconfig_file" {
+  template = file("./kubeconfig.tpl")
+  vars = {
+    CLUSTER_NAME : var.eks_cluster_context_name
+    CA_DATA : data.aws_eks_cluster.testing_cluster.certificate_authority[0].data
+    SERVER_ENDPOINT : data.aws_eks_cluster.testing_cluster.endpoint
+    TOKEN = data.aws_eks_cluster_auth.testing_cluster.token
+  }
+}
+
+resource "local_file" "kubeconfig" {
+  content = data.template_file.kubeconfig_file.rendered
+  filename = "${var.kube_directory_path}/config"
+}
+
+### Setting up the sample app on the cluster
+
+resource "kubernetes_deployment" "dotnet_app_deployment" {
+
+  metadata {
+    name      = "dotnet-app-deployment-${var.test_id}"
+    namespace = var.test_namespace
+    labels    = {
+      app = "dotnet-app"
+    }
+  }
+
+  spec {
+    replicas = 1
+    selector {
+      match_labels = {
+        app = "dotnet-app"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          app = "dotnet-app"
+        }
+        annotations = {
+          # these annotations allow for OTel Dotnet instrumentation
+          "instrumentation.opentelemetry.io/inject-dotnet": "true"
+        }
+      }
+      spec {
+        service_account_name = var.service_account_aws_access
+        container {
+          name = "back-end"
+          image = var.dotnet_app_image
+          image_pull_policy = "Always"
+          env {
+              #inject the test id to service name for unique App Signals metrics
+              name = "OTEL_SERVICE_NAME"
+              value = "dotnet-application-${var.test_id}"
+            }
+          env {
+              #Updated plugin name since it's incorrect in the addon
+              name = "OTEL_DOTNET_AUTO_PLUGINS"
+              value = "AWS.Distro.OpenTelemetry.AutoInstrumentation.Plugin, AWS.Distro.OpenTelemetry.AutoInstrumentation"
+            }
+          env {
+              #Updated plugin name since it's incorrect in the addon
+              name = "OTEL_EXPORTER_OTLP_ENDPOINT"
+              value = " http://cloudwatch-agent.amazon-cloudwatch:4316"
+            }
+          
+          port {
+            container_port = 8080
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "kubernetes_service" "dotnet_app_service" {
+  depends_on = [ kubernetes_deployment.dotnet_app_deployment ]
+
+  metadata {
+    name = "dotnet-app-service"
+    namespace = var.test_namespace
+  }
+  spec {
+    type = "NodePort"
+    selector = {
+        app = "dotnet-app"
+    }
+    port {
+      protocol = "TCP"
+      port = 8080
+      target_port = 8080
+      node_port = 30100
+    }
+  }
+}
+
+resource "kubernetes_ingress_v1" "dotnet-app-ingress" {
```

Comment:
Harry is working on removing public endpoint currently, can you sync with him on this? We want to make the sample app generate traffic itself and have the validator check the data without relying on calling the application.
This caused our current security sev2, let's avoid the same mistake.